src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,321 @@
+        final PrintWriter writer = new PrintWriter(osw, false);
+
+        try {
+            for (DetailAST top = JavaParser.parseFile(new File(inputFile),
+                    JavaParser.Options.WITH_COMMENTS); top != null; top = top.getNextSibling()) {
+                if (top.getType() != TokenTypes.CLASS_DEF) {
+                    continue;
+                }
+                // OBJBLOCK is always the last child of CLASS_DEF.
+                final DetailAST objBlock = top.getLastChild();
+                for (DetailAST ast = objBlock.getFirstChild(); ast != null;
+                        ast = ast.getNextSibling()) {
+                    if (isPublicStaticFinalIntField(ast)) {
+                        final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
+                        final String firstJavadocSentence = getFirstJavadocSentence(modifiers);
+                        if (firstJavadocSentence != null) {
+                            writer.println(getName(ast) + "=" + firstJavadocSentence.trim());
+                        }
+                    }
+                }
+            }
+        }
+        finally {
+            writer.close();
+        }
+    }
+
+    /**
+     * Checks that the DetailAST is a {@code public} {@code static} {@code final} {@code int} field.
+     * @param ast to process
+     * @return {@code true} if matches; {@code false} otherwise
+     */
+    private static boolean isPublicStaticFinalIntField(DetailAST ast) {
+        boolean result = ast.getType() == TokenTypes.VARIABLE_DEF;
+        if (result) {
+            final DetailAST type = ast.findFirstToken(TokenTypes.TYPE);
+            result = type.getFirstChild().getType() == TokenTypes.LITERAL_INT;
+            if (result) {
+                final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
+                result = modifiers.findFirstToken(TokenTypes.LITERAL_PUBLIC) != null
+                    && modifiers.findFirstToken(TokenTypes.LITERAL_STATIC) != null
+                    && modifiers.findFirstToken(TokenTypes.FINAL) != null;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Extracts the name of an ast.
+     * @param ast to extract the name
+     * @return the text content of the inner {@code TokenTypes.IDENT} node
+     */
+    private static String getName(DetailAST ast) {
+        return ast.findFirstToken(TokenTypes.IDENT).getText();
+    }
+
+    /**
+     * Extracts the first sentence of the comment of an DetailAST.
+     * @param ast to extract the first sentence
+     * @return the first sentence of the inner {@code TokenTypes.BLOCK_COMMENT_BEGIN} node
+     *      or {@code null} if the first sentence is absent or malformed (does not end with period)
+     * @throws CheckstyleException if a javadoc comment can not be parsed or an unsupported inline
+     *      tag found
+     */
+    private static String getFirstJavadocSentence(DetailAST ast) throws CheckstyleException {
```

Comment:
We have special Check for validation of first sentence, can we reuse logic from it ?

Reply:
Originally it was. But Richard proposed to get rid of regular expressions. See
https://github.com/checkstyle/checkstyle/pull/5399#discussion_r158961170

Reply:
ok, looks like we get first sentence in special way.